File: backend/locator-api/src/main/java/com/mype/locator_api/model/Mype.java
package com.mype.locator_api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "mypes")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Mype {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(name = "nombre_comercial", nullable = false)
    private String nombreComercial;

    @NotBlank
    @Column(unique = true, nullable = false)
    private String ruc;

    private String direccion;

    @Column(precision = 10, scale = 8)
    private BigDecimal latitud;

    @Column(precision = 11, scale = 8)
    private BigDecimal longitud;

    @OneToOne
    @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNombreComercial() {
        return nombreComercial;
    }

    public void setNombreComercial(String nombreComercial) {
        this.nombreComercial = nombreComercial;
    }

    public String getRuc() {
        return ruc;
    }

    public void setRuc(String ruc) {
        this.ruc = ruc;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public BigDecimal getLatitud() {
        return latitud;
    }

    public void setLatitud(BigDecimal latitud) {
        this.latitud = latitud;
    }

    public BigDecimal getLongitud() {
        return longitud;
    }

    public void setLongitud(BigDecimal longitud) {
        this.longitud = longitud;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }


    
}

File: mype-frontend/src/app/services/mype.ts
export interface Usuario {
  id: number;
  nombreUsuario: string;
  correo: string;
  rol: string;
}

export interface Mype {
  id: number;
  nombreComercial: string;
  ruc: string;
  direccion: string;
  latitud: number;
  longitud: number;
  usuario?: Usuario;
}

export interface CrearUsuarioRequest {
  nombreUsuario: string;
  contrasenia: string;
  correo: string;
  rol: string;
}

export interface CrearMypeRequest {
  nombreComercial: string;
  ruc: string;
  direccion: string;
  latitud: number | null;
  longitud: number | null;
  usuario: { id: number };
}

export interface LoginRequest {
  nombreUsuario: string;
  contrasenia: string;
}

export interface AuthResponse {
  tokenAcceso: string;
  nombreUsuario: string;
  rol: string;
}